controllers/gateway_creator.go

```diff
@@ -0,0 +1,20 @@
+package controllers
```

Comment:
If we move to the logic from the controller to the processor we won't need those files

Reply:
I agree. I wrote this stupid wrapper, because I was trying to assign one interface to another, and Go would not allow that. We can mitigate that by just using the gateway interfaces coming from `processors` everywhere.

Reply:
See above comment